server/py/framework/utils/clients/chief.py

```diff
@@ -13,24 +13,22 @@
 # limitations under the License.
 #
 import contextlib
-import copy
-import http.cookies
 import typing
-import urllib.parse
 
 import aiohttp
 import fastapi
 
 import mlrun.common.schemas
 import mlrun.errors
 import mlrun.utils.singleton
-from mlrun.utils import logger
+
+import framework.utils.clients.messaging
 
 
 # we were thinking to simply use httpdb, but decided to have a separated class for simplicity for now until
 # this class evolves, but this should be reconsidered when adding more functionality to the class
 class Client(
-    metaclass=mlrun.utils.singleton.AbstractSingleton,
+    framework.utils.clients.messaging.BaseClient,
```

Comment:
I think that rather chief implement messaging client, it should hold that as a dependency, and the reason is that chief client is not a messaging client but rather holds it + state upon need.